node/src/Pos/Explorer/DB.hs

```diff
@@ -104,10 +167,86 @@ prepareExplorerDB = do
             addressCoinPairs = utxoToAddressCoinPairs utxo
         putGenesisBalances addressCoinPairs
         putInitFlag
+
     -- Smooth migration for CSE-228.
     unlessM utxoSumInitializedM $ do
         putCurrentUtxoSum
 
+    -- Smooth migration for CSE-236.
+    unlessM blockEpochPagesInitializedM $ do
+        convertOldEpochBlocksFormat
```

Comment:
Is this really the best place to put the migration? This is in a query. What is to stop this happening multiple times concurrently? Surely it is better to do the migration as part of server startup and opening the DB and before starting the web server listening thread.

From the other comments here, I am not clear about whether this migration is enough or if devops need to do anything with clearing the DB.

Reply:
@dcoutts this is not in a query. This code is within `prepareExplorerDB` which is executed at server startup

Reply:
Sorry, I was mislead by the github diff, it looked like it was inside `getLastTransactions`. Oops! Ok, never mind about that complaint then.

Reply:
AFAIK they don't have to do anything, the migration will run only on those machines that have the "old" DB epoch format when preparing explorer database.